### PR TITLE
Adds lobby music

### PR DESCRIPTION
**Pull request by ike709**
## About the PR 
Adds lobby music. Can be toggled in the settings or directly via cvar. Also fixes ambient hum so it actually stops when you disconnect.

Also renames AmbientSystem to BackgroundAudioSystem.

TODO:
- [x] Add a couple more songs
- [x] Consider renaming AmbienceSystem?
- [x] Finalize the audio params. I'm currently testing with terrible earbuds.

**Screenshots**

https://streamable.com/v9yeg9

**Changelog**

:cl:
- add: Added lobby music. It can be disabled in the audio settings.
- fix: Fixed the ambience continuing to play after you disconnect.

### CODE REVIEW

File: Content.Client/GameObjects/EntitySystems/AmbienceSystem.cs [hunk 1]
```diff
@@ -90,6 +134,35 @@ private void EndAmbience()
...
             _ambientStream = null;
         }
+
+        private void LobbyMusicCVarChanged(bool musicEnabled)
+        {
+            if (!musicEnabled)
+            {
+                EndLobbyMusic();
+            }
+            else if (_stateManager.CurrentState is LobbyState)
```
Comment (GraniteSidewalk) [inline Content.Client/GameObjects/EntitySystems/AmbienceSystem.cs:144]: You forgot the case where musicEnabled && !_stateManager.CurrentState is LobbyState - might want to call EndLobbyMusic() in that case just to be safe

**Comment by GraniteSidewalk** [comment]
Pretty simple PR, don't see really anything wrong with it but I do think you should rename AmbienceSystem to something better.

**Comment by ike709** [comment]
I can't test the audio params super well so I'll wait for codejannie feedback.

File: Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            {
+                EndLobbyMusic();
+            }
+        }
+
+        private void StartLobbyMusic()
+        {
+            EndLobbyMusic();
+            var file = _clientGameTicker.LobbySong;
+            if (file == null) // Just in case we fail to get a song from the server, play one at random.
```
Comment (ike709) [inline Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs:162]: For some reason we're failing to get the server's song and we're falling back on a randomized one.

Comment (ike709) [reply Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs:162]: If I disconnect and then reconnect, all connections after the first work fine.

Comment (ike709) [reply Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs:162]: The StateChanged code seems to fire before the client actually receives the net message. Not sure what to do about that.

**Comment by ike709** [comment]
Code should be good to go, I think?

**Comment by ike709** [comment]
Should be good to go?

**Comment by ike709** [comment]
One bug: Latejoins don't get lobby music.

**Comment by ike709** [comment]
I have no idea why latejoins don't get music.

**Comment by ike709** [comment]
When latejoining (and _only_ when latejoining, somehow), `MsgTickerLobbyStatus` is sent containing the correct value but when it is received the song string is null.

**Comment by ike709** [comment]
Latejoin music fixed, this should be fully good to go.

File: Content.Shared/GameTicking/SharedGameTicker.cs [hunk 3]
```diff
@@ -96,9 +96,12 @@ protected class MsgTickerLobbyStatus : NetMessage
             public TimeSpan StartTime { get; set; }
             public bool Paused { get; set; }
 
+            public string? LobbySong { get; set; }
+
```
Comment (metalgearsloth) [inline Content.Shared/GameTicking/SharedGameTicker.cs:100]: Just for readability sake probably good to have property order in buffer order.

File: Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            _lobbyStream = _audioSystem.Play(file, _lobbyParams);
+        }
+
+        private void EndLobbyMusic()
+        {
+            if (_lobbyStream == null)
+            {
+                return;
+            }
+            _lobbyStream.Stop();
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs:190]:
```suggestion
            _lobbyStream?.Stop();
```

File: Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            _ambientStream = _audioSystem.Play(file, _ambientParams);
+        }
+
+        private void EndAmbience()
+        {
+            if (_ambientStream == null)
+            {
+                return;
+            }
+            _ambientStream.Stop();
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/EntitySystems/BackgroundAudioSystem.cs:142]:
```suggestion
            _ambientStream?.Stop();
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Seems okay if it works ingame.

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**